### ISSUE TITLE
RealmChangeListener not called when setting null value

### ISSUE THREAD

**Issue by huibvv** [realm/realm-java#2366]
When setting the value of a field to null, the RealmChangeListener is not being called. Setting a empty string (or any other value for that matter) does trigger the notification

**Comment by beeender** [comment]
Can you please share your code which can reproduce this? I doubt you were setting `null` on a standalone object which won't change the Realm and trigger the notification.

**Comment by huibvv** [comment]
`Realm realm = Realm.getDefaultInstance();
TriggerAction action = realm.where(TriggerAction.class).findFirst();
HardwareActorProperty actor = action.getProperties().where().equalTo("Name", "name").findFirst();
            
    realm .beginTransaction();
            actor.setValue(null);
            realm .commitTransaction();`

The item being updated is a child record, where the value needs to be set to null. 

The objects classes are seupt as following
`public class TriggerAction extends RealmObject {
    @PrimaryKey
    private transient String refId;

    private String Id;
    private transient long delayTime;
    private transient long durationTime;

    private RealmList<HardwareActorProperty> Properties;

    public String getId() {
        return Id;
    }

    public void setId(String id) {
        Id = id;
    }

    public RealmList<HardwareActorProperty> getProperties() {
        return Properties;
    }

    public void setProperties(RealmList<HardwareActorProperty> properties) {
        Properties = properties;
    }

    public long getDelayTime() {
        return delayTime;
    }

    public void setDelayTime(long delayTime) {
        this.delayTime = delayTime;
    }

    public long getDurationTime() {
        return durationTime;
    }

    public void setDurationTime(long durationTime) {
        this.durationTime = durationTime;
    }

    public String getRefId() {
        return refId;
    }

    public void setRefId(String refId) {
        this.refId = refId;
    }
}
`
`public class HardwareActorProperty extends RealmObject{
    private String refId;
    private String Name;
    private String Value;
    private String ValueAfterTimer;

    public String getRefId() {
        return refId;
    }

    public void setRefId(String refId) {
        this.refId = refId;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public String getValue() {
        return Value;
    }

    public void setValue(String value) {
        Value = value;
    }

    public String getValueAfterTimer() {
        return ValueAfterTimer;
    }

    public void setValueAfterTimer(String valueAfterTimer) {
        ValueAfterTimer = valueAfterTimer;
    }
}`

**Comment by cmelchior** [comment]
I could reproduce the issue with the following unit test:

```
    @Test
    @RunTestInLooperThread
    public void setter_nullTriggersChangesListener() {
        Realm realm = looperThread.realm;
        realm.beginTransaction();
        final Cat obj = realm.createObject(Cat.class);
        obj.setName("Foo");
        realm.commitTransaction();
        assertEquals("Foo", obj.getName());

        final RealmResults<Cat> cats = realm.allObjects(Cat.class);
        cats.addChangeListener(new RealmChangeListener() {
            @Override
            public void onChange() {
                assertEquals(null, cats.get(0).getName());
                looperThread.testComplete();
            }
        });
        
        realm.beginTransaction();
        obj.setName(null);
        realm.commitTransaction();
    }
```

I could nether trigger RealmObject nor RealmResults change listeners which indicate that core doesn't update the table version when a value is set to null. We need to look further into this to verify this.

**Comment by beeender** [comment]
Fixed in core, will close this issue when core released.

**Comment by beeender** [comment]
This is fixed by #2397 , will be released in 0.88
